Utils/src/PrescaleWeightProducer.cc

```diff
@@ -0,0 +1,203 @@
+// -*- C++ -*-
+//
+// Package:    PrescaleWeightProducer
+// Class:      PrescaleWeightProducer
+//
+/**\class PrescaleWeightProducer PrescaleWeightProducer.cc AllHadronicSusy/Utils/src/PrescaleWeightProducer.cc
+   Description: <one line class summary>
+   Implementation:
+   <Notes on implementation>
+*/
+//
+// Created by Sam Bein, Tue Sept 10 17:58:04 CET 2015
+// $Id: PrescaleWeightProducer.cc,v 1.7 2012/07/09 11:43:55 Bein Exp $
+
+// system include files
+#include <cmath>
+#include <memory>
+#include <string>
+#include <iostream>
+#include <vector>
+#include <map>
+#include <stdlib.h> 
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "HLTrigger/HLTcore/interface/HLTConfigProvider.h"
+#include "FWCore/Common/interface/TriggerNames.h"
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "FWCore/Common/interface/TriggerResultsByName.h"
+#include "DataFormats/HLTReco/interface/TriggerEvent.h"
+#include "DataFormats/PatCandidates/interface/PackedTriggerPrescales.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+
+using namespace std;
+using namespace trigger;
+using namespace edm;
+
+
+class PrescaleWeightProducer: public edm::EDProducer {
+public:
+  explicit PrescaleWeightProducer(const edm::ParameterSet&);
+  ~PrescaleWeightProducer();
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);//Van
+
+private:
+  virtual void produce(edm::Event&, const edm::EventSetup&);
+  virtual void beginJob();
+  virtual void endJob();
+  virtual void beginRun(edm::Run&, edm::EventSetup const&);
+  const double _startWeight;
+  const double _LumiScale;
+  const double _PrescaleCut;
+  const bool _PFHTWeights;
+  edm::InputTag _weightName;
+  string _processName;
+
+  edm::EDGetTokenT<edm::TriggerResults> _triggerBits;
+  edm::EDGetTokenT<pat::TriggerObjectStandAloneCollection> _triggerObjects;
+  edm::EDGetTokenT<pat::PackedTriggerPrescales> _triggerPrescales;
+};
+
+PrescaleWeightProducer::PrescaleWeightProducer(const edm::ParameterSet& iConfig) :
+  _startWeight(iConfig.getParameter<double> ("weight")), _LumiScale(iConfig.getParameter<double> ("LumiScale")), _PrescaleCut(iConfig.getParameter<double> ("PrescaleCut")), _PFHTWeights(iConfig.getParameter<bool> ("PFHTWeights")), _weightName(iConfig.getParameter<edm::InputTag> ("weightName")), _processName(iConfig.getParameter<string> ("HLTProcess")), _triggerBits(consumes<edm::TriggerResults>(iConfig.getParameter<edm::InputTag>("bits"))), _triggerObjects(consumes<pat::TriggerObjectStandAloneCollection>(iConfig.getParameter<edm::InputTag>("objects"))), _triggerPrescales(consumes<pat::PackedTriggerPrescales>(iConfig.getParameter<edm::InputTag>("prescales"))) 
```

Comment:
Can you split this initialization list into a few lines so it doesn't stretch toward infinity?